src/nvim/option.c

```diff
@@ -908,68 +908,194 @@ static void munge_string_opt_val(char **varp, char **oldval, char **const origva
   }
 }
 
-/// Part of do_set() for string options.
-static void do_set_string(int opt_idx, int opt_flags, char **argp, int nextchar, set_op_T op_arg,
-                          uint32_t flags, char *varp_arg, char *errbuf, size_t errbuflen,
-                          int *value_checked, const char **errmsg)
+/// Get the default value for a string option.
+static char *stropt_get_default_val(int opt_idx, uint64_t flags)
+{
+  char *newval = options[opt_idx].def_val;
+  // expand environment variables and ~ since the default value was
+  // already expanded, only required when an environment variable was set
+  // later
+  if (newval == NULL) {
+    newval = empty_option;
+  } else if (!(options[opt_idx].flags & P_NO_DEF_EXP)) {
+    char *s = option_expand(opt_idx, newval);
+    if (s == NULL) {
+      s = newval;
+    }
+    newval = xstrdup(s);
+  } else {
+    newval = xstrdup(newval);
+  }
+  return newval;
+}
+
+// Copy the new string into allocated memory.
+// Can't use set_string_option_direct(), because we need to remove the
+// backslashes.
```

Comment:
```suggestion
/// Copy the new string value into allocated memory for the option.
/// Can't use set_string_option_direct(), because we need to remove the
/// backslashes.
```